develop/devguide/Automation/Howto/Creating_a_cypress_test_for_an_interactive_automation_script.md

```diff
@@ -0,0 +1,231 @@
+          // This is determined by a simple string search looking for "engine.ShowUI" in the source code.
+          // However, due to the NuGet package, this string can no longer be detected.
+          // This comment is here as a temporary workaround until it has been fixed.
+          //// engine.ShowUI(
+
+          try
+          {
+             // Controls the event loop and switch between dialogs
+             var controller = new InteractiveController(engine);
+
+             // Create an instance of the dialog you wish to show.
+             var helloWorldDialog = new HelloWorldDialog(engine);
+             
+             // Starts the event loop and shows the first dialog.
+             controller.ShowDialog(helloWorldDialog);
+          }
+          catch (ScriptAbortException)
+          {
+             // Catch normal abort exceptions (engine.ExitFail or engine.ExitSuccess)
+             throw; // Comment if it should be treated as a normal exit of the script.
+          }
+          catch (ScriptForceAbortException)
+          {
+             // Catch forced abort exceptions, caused via external maintenance messages.
+             throw;
+          }
+          catch (ScriptTimeoutException)
+          {
+             // Catch timeout exceptions for when a script has been running for too long.
+             throw;
+          }
+          catch (InteractiveUserDetachedException)
+          {
+             // Catch a user detaching from the interactive script by closing the window.
+             // Only applicable for interactive scripts, can be removed for non-interactive scripts.
+             throw;
+          }
+          catch (Exception e)
+          {
+             engine.ExitFail($"Run|Something went wrong: {e}");
+          }
+       }
+   }
+
+   public class HelloWorldDialog : Dialog
+   {
+      public HelloWorldDialog(IEngine engine) : base(engine)
+      {
+          // Create a label widget
+          var label = new Label("Hello, World!") { Style = TextStyle.Title };
+          
+          // Define the debug tag to be used in the cypress test
+          label.DebugTag = "label";
+          
+          // Add it to the dialog grid at position 0,0
+          AddWidget(label, 0, 0);
+          
+          // Create a button widget
+          var button = new Button("OK");
+          
+          // Define the debug tag to be used in the cypress test
+          button.DebugTag = "button";
+          
+          // Add it to the dialog just below the label
+          AddWidget(button, 1, 0);
+          
+          // Add an action to be performed whenever the button is clicked
+          button.Pressed += (sender, args) => engine.ExitSuccess("Done");
+      }
+    }
+    ```
+
+## Setting up the web environment
+
+1. [Create a new dashboard](xref:Creating_a_completely_new_dashboard).
+
+1. Add a button to the dashboard and configure it to trigger the created interactive Automation script when clicked.
```

Comment:
There's a problem with the procedure at this point: Even when the _ReportsAndDashboardsButton_ soft-launch option is enabled, it is not possible to add a button that links directly to an Automation script in the most recent version of the Dashboards app. The button has to be linked to a button parameter instead, but I don't think this is what you have in mind here. Would it be an option to change this part of your procedure to use a low-code app instead? Then adding a button that links directly to a script will be no problem.

Reply:
To use an automation script in a dashboard you'd also need the _ReportsAndDashboardsAutomationScript_ soft-launch option. So indeed, an LCA might be a better option.

Reply:
Thanks for your remarks! I've changed it to use a low-code app instead as suggested!